### DISCUSSION TITLE
Editing a webradio url leads to a failure on the home screen

### DISCUSSION THREAD

**Discussion by stefanroellin** [jcorporation/myMPD#687] [Q&A] [answered]
myMPD version: 9.1.1

**Describe the bug**

I have changed a url of a webradio favourite, navigated to the home screen and clicked the corresponding icon there, which lead to a failure.

**To Reproduce**

1. Add a webradio, e.g. "SRF 1" from Switzerland, to the favourites.
2. Add the same radio to the home screen.
2. In the favourites, edit the url, i.e. change it from http://streaming.swisstxt.ch/m/drs1/mp3_128
    to http://stream.srg-ssr.ch/m/regi_zentr/mp3_128 .
3. Navigate to the home screen.
4. Try to replace and play the webradio. 
This leads to the error "Error in response to command: mympd_api_queue_replace_with_playlist".

**Expected behavior**

I would expect that the webradio still works either with the original url or with the updated url, which is currently not the case.

**Comment by jcorporation** [comment, marked as answer]
Webradio favorites are saved as files. The filename is derived from the webradio url. If you change the webradio url, you implicitly change the filename. The home icon then references a not existent webradio favorite file.

This is not a bug, its by design. The same behavior exists for home icons that references directories, playlists, etc.

**Reply by stefanroellin** [reply]
Thanks for the clarification. I had the feeling the behaviour could be as designed.

And also thanks a lot for the amazing software!
